genesis-volatility/test/adapter_test.js

```diff
@@ -0,0 +1,55 @@
+const assert = require('chai').assert
+const createRequest = require('../adapter').createRequest
+
+describe('createRequest', () => {
+  const jobID = '1'
+
+  context('successful calls', () => {
+    const requests = [
+      { name: 'id not supplied', testData: { data: { symbol: 'ETH', result: 'oneDayIv' } } },
+      { name: '1 day', testData: { id: jobID, data: { symbol: 'ETH', result: 'oneDayIv' } } },
+      { name: '2 days', testData: { id: jobID, data: { symbol: 'ETH', result: 'twoDayIv' } } },
+      { name: '1 week', testData: { id: jobID, data: { symbol: 'ETH', result: 'sevenDayIv' } } },
+      { name: '2 weeks', testData: { id: jobID, data: { symbol: 'ETH', result: 'fourteenDayIv' } } },
+      { name: '3 weeks', testData: { id: jobID, data: { symbol: 'ETH', result: 'twentyOneDayIv' } } },
+      { name: '4 weeks', testData: { id: jobID, data: { symbol: 'ETH', result: 'twentyEightDayIv' } } },
+      { name: '1 day BTC', testData: { id: jobID, data: { symbol: 'BTC', result: 'oneDayIv' } } }
+    ]
+
+    requests.forEach(req => {
+      it(`${req.name}`, (done) => {
+        createRequest(req.testData, (statusCode, data) => {
+          assert.equal(statusCode, 200)
+          assert.equal(data.jobRunID, jobID)
+          assert.isNotEmpty(data.data)
+          assert.isAbove(data.result, 0)
+          assert.isAbove(data.data.result, 0)
+          done()
+        })
+      })
+    })
+  })
+
+  context('error calls', () => {
```

Comment:
Please split this into two tests, one that fails on input validation and the other integration test, where the call goes all the way to the server.

This is to make it easier to merge with changes we have in the private repo.